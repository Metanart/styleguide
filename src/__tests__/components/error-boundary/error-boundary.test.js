import React from 'react';
import { create, act } from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import sinon from 'sinon';
import ErrorBoundary from '../../../components/error-boundary/error-boundary';

class BrokenComponent extends React.PureComponent {
    render() {
        return (
            <div>
                <input type="text" value={null} />
            </div>
        );
    }
}

describe('ErrorBoundary tests:', () => {
    it('ErrorBoundary render no errors', () => {
        let root;

        act(() => {
            root = create(
                <ErrorBoundary>
                    <div>I am valid</div>
                </ErrorBoundary>
            );
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('ErrorBoundary render error', () => {
        let root;

        act(() => {
            root = create(
                <ErrorBoundary>
                    <BrokenComponent />
                </ErrorBoundary>
            );
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('ErrorBoundary getDerivedStateFromError test', () => {
        function Something() {
            return null;
        }
        const spy = sinon.spy();
        const wrapper = shallow(
            <ErrorBoundary spy={spy}>
                <Something />
            </ErrorBoundary>
        );
        const error = new Error('This is a test error!');
        wrapper.find(Something).simulateError(error);

        expect(wrapper.state()).toHaveProperty('error');
        expect(wrapper.state().error.toString()).toBe('Error: This is a test error!');

        expect(wrapper.html()).toMatchSnapshot();

        wrapper.instance().toggleErrorStack();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
