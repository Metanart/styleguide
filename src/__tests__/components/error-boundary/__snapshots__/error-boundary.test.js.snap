// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary tests: ErrorBoundary getDerivedStateFromError test 1`] = `"<section class=\\"sc-AxjAm cgoQWF\\"><h3 class=\\"sc-AxirZ fwrrpd\\">Error: This is a test error!</h3><p class=\\"sc-AxiKw cWdryN\\">Show error stack</p></section>"`;

exports[`ErrorBoundary tests: ErrorBoundary getDerivedStateFromError test 2`] = `
"<section class=\\"sc-AxjAm cgoQWF\\"><h3 class=\\"sc-AxirZ fwrrpd\\">Error: This is a test error!</h3><p class=\\"sc-AxiKw cWdryN\\">Hide error stack</p><p class=\\"sc-AxhCb gOihBj\\">Error: This is a test error!
    at Object.it (/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/src/__tests__/components/error-boundary/error-boundary.test.js:58:23)
    at Object.asyncJestTest (/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)
    at resolve (/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/node_modules/jest-jasmine2/build/queueRunner.js:45:12)
    at new Promise (&lt;anonymous&gt;)
    at mapper (/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/node_modules/jest-jasmine2/build/queueRunner.js:28:19)
    at promise.then (/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/node_modules/jest-jasmine2/build/queueRunner.js:75:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)</p></section>"
`;

exports[`ErrorBoundary tests: ErrorBoundary render error 1`] = `
<div>
  <input
    type="text"
    value={null}
  />
</div>
`;

exports[`ErrorBoundary tests: ErrorBoundary render no errors 1`] = `
<div>
  I am valid
</div>
`;
